nucypher/cli/worklock.py

```diff
@@ -0,0 +1,210 @@
+
+option_bidder_address = click.option('--bidder-address',
+                                     help="Bidder's checksum address.",
+                                     type=EIP55_CHECKSUM_ADDRESS)
+
+
+def _setup_emitter(general_config):
+    emitter = general_config.emitter
+    emitter.clear()
+    emitter.banner(WORKLOCK_BANNER)
+    return emitter
+
+
+class WorkLockOptions:
+
+    __option_name__ = 'worklock_options'
+
+    def __init__(self, bidder_address: str):
+        self.bidder_address = bidder_address
+
+    def create_agent(self, registry):
+        agent = ContractAgency.get_agent(WorkLockAgent, registry=registry)
+        return agent
+
+
+group_worklock_options = group_options(
+    WorkLockOptions,
+    bidder_address=option_bidder_address)
+
+
+@click.group()
+def worklock():
+    """
+    Manage stakes and other staker-related operations.
+    """
+    pass
+
+
+@worklock.command()
+@group_registry_options
+@group_worklock_options
+@group_general_config
+def status(general_config, registry_options, worklock_options):
+    emitter = _setup_emitter(general_config)
+    registry = registry_options.get_registry(emitter, general_config.debug)
+    paint_worklock_status(emitter=emitter, registry=registry)
+    if worklock_options.bidder_address:
+        paint_worklock_participant_status(emitter=emitter,
+                                          registry=registry,
+                                          bidder_address=worklock_options.bidder_address)
+    return  # Exit
+
+
+@worklock.command()
+@option_force
+@group_registry_options
+@group_worklock_options
+@group_general_config
+@click.option('--value', help="Eth value of bid", type=click.INT)
+def bid(general_config, worklock_options, registry_options, force, value):
+    emitter = _setup_emitter(general_config)
+    if not value:
+        if force:
+            raise click.MissingParameter("Missing --value.")
+        value = int(Web3.fromWei(click.prompt("Enter bid amount in ETH", type=click.FloatRange(min=0)), 'wei'))
+
+    if not worklock_options.bidder_address:
+        worklock_options.bidder_address = select_client_account(emitter=emitter,
+                                                                provider_uri=general_config.provider_uri)
+    registry = registry_options.get_registry(emitter, general_config.debug)
+    worklock_agent = worklock_options.create_agent(registry=registry)
+
+    if not force:
+        paint_worklock_participant_notice(emitter=emitter, bidder_address=worklock_options.bidder_address, registry=registry)
+        click.confirm(f"Place WorkLock bid of {Web3.fromWei(value, 'ether')} ETH?", abort=True)
+
+    receipt = worklock_agent.bid(bidder_address=worklock_options.bidder_address, value=value)
+    emitter.message("Publishing WorkLock Bid...")
+    paint_receipt_summary(receipt=receipt, emitter=emitter, chain_name=worklock_agent.blockchain.client.chain_name)
+    return  # Exit
+
+
+@worklock.command()
```

Comment:
Because of the underscore vs dash, the name needs to be defined.

```suggestion
@worklock.command(name='cancel-bid')
```

Reply:
ignore.